### ISSUE TITLE
how to use pretrained model of Attention-92(MX)

### ISSUE THREAD

**Issue by leoluopy** [JDAI-CV/FaceX-Zoo#28]
i have downloaded the pretrained model : Attention-92(MX)
but how to use it .
I have tried the following code:
`
model = ResidualAttentionNet(1, 1, 1, 512, 7, 7)
model.load_state_dict(torch.load("/home/leo/workspace/pretrained/Attention92/Epoch_17.pt"))
`

and it failed with the error below:
`{RuntimeError}Error(s) in loading state_dict for ResidualAttentionNet:
	Missing key(s) in state_dict: "conv1.0.weight", "conv1.1.weight", "conv1.1.bias", "conv1.1.running_mean", "conv1.1.running_var", "attention_body.0.bn1.weight", "attention_body.0.bn1.bias", "attention_body.0.bn1.running_mean", "attention_body.0.bn1.running_var", "attention_body.0.conv1.weight", "attention_body.0.bn2.weight", "attention_body.0.bn2.bias", "attention_body.0.bn2.running_mean", "attention_body.0.bn2.running_var", "attention_body.0.conv2.weight", "attention_body.0.bn3.weight", "attention_body.0.bn3.bias", "attention_body.0.bn3.running_mean", "attention_body.0.bn3.running_var", "attention_body.0.conv3.weight", "attention_body.0.conv4.weight", "attention_body.1.first_residual_blocks.bn1.weight", "attention_body.1.first_residual_blocks.bn1.bias", "attention_body.1.first_residual_blocks.bn1.running_mean", "attention_body.1.first_residual_blocks.bn1.running_var", "attention_body.1.first_residual_blocks.conv1.w...`

can anyone help to solve this problem and make use of the pretrained model

**Comment by wang21jun** [comment]
Please refer to ModelLoader in "FaceX-Zoo/test_protocol/utils/model_loader.py "
    def load_model(self, model_path):
        """The custom method to load a model.                                                                                                                                                                                                                                                                    
        Args:                                                                                                                                                                                                                                                                              
            model_path(str):                                                                                                                                                                                                                                                                             
        Returns:                                                                                                                                                                                                                                                                           
            model(object): initialized model.                                                                                                                                                                                                                                              
        """
        model_dict = self.model.state_dict()
        pretrained_dict = torch.load(model_path)['state_dict']                                                                                                                                                                                                                                    
        new_pretrained_dict = {}
        for k in model_dict:
            new_pretrained_dict[k] = pretrained_dict['backbone.'+k] # tradition training                                                                                                                                                                                                                    
        model_dict.update(new_pretrained_dict)
        self.model.load_state_dict(model_dict)
        model = torch.nn.DataParallel(self.model).cuda()
        return model

**Comment by wang21jun** [comment]
Please refer to ModelLoader in "FaceX-Zoo/test_protocol/utils/model_loader.py "
```python
    def load_model(self, model_path):
        """The custom method to load a model.                                                                                                                                                                                                                                              
                                                                                                                                                                                                                                                                                           
        Args:                                                                                                                                                                                                                                                                              
            model_path(str): the path of the weight file.                                                                                                                                                                                                                                  
                                                                                                                                                                                                                                                                                           
        Returns:                                                                                                                                                                                                                                                                           
            model(object): initialized model.                                                                                                                                                                                                                                              
        """
        model_dict = self.model.state_dict()
        pretrained_dict = torch.load(model_path)['state_dict']                                                                                                                                                                                                                                        
        new_pretrained_dict = {}
        for k in model_dict:
            new_pretrained_dict[k] = pretrained_dict['backbone.'+k] # tradition training                                                                                                                                                                                                                        
        model_dict.update(new_pretrained_dict)
        self.model.load_state_dict(model_dict)
        model = torch.nn.DataParallel(self.model).cuda()
        return model
```

**Comment by leoluopy** [comment]
thanks to @wang21jun 
and after walking through the code of FaceX , i found this is a simple way to use the pretrained model . code is available below:
`

class FaceModel(torch.nn.Module):
    """Define a traditional face model which contains a backbone and a head.

    Attributes:
        backbone(object): the backbone of face model.
        head(object): the head of face model.
    """

    def __init__(self):
        """Init face model by backbone factorcy and head factory.

        Args:
            backbone_factory(object): produce a backbone according to config files.
            head_factory(object): produce a head according to config files.
        """
        super(FaceModel, self).__init__()
        self.backbone = ResidualAttentionNet(1, 2, 3, 512, 7, 7)
        self.head = ArcFace(512, 72778, 0.35, 0.0, 32)

    def forward(self, data, label):
        feat = self.backbone.forward(data)
        pred = self.head.forward(feat, label)
        return pred

    def forward(self, data):
        feat = self.backbone.forward(data)
        return feat


if __name__ == '__main__':
    model = FaceModel()
    modelPath = "/home/leo/workspace/pretrained/Attention92/Epoch_17.pt"
    model.load_state_dict(torch.load(modelPath)['state_dict'])
    if torch.cuda.is_available() is True:
        model = torch.nn.DataParallel(model).cuda()
    model.eval()

    input = torch.rand([1, 3, 112, 112], dtype=torch.float32).cuda()
    out = model(input)

    print(out.shape)
    print("END")
`
